executable-spec/src/Cardano/Ledger/Spec/STS/Update/Ideation.hs

```diff
@@ -84,6 +84,10 @@ data St hashAlgo
     , wrsips :: !(Map (Data.SIPHash hashAlgo) Slot)
       -- ^ When a SIP was revealed
 
+    , sipdb :: !(Map (Data.SIPHash hashAlgo) (SIP hashAlgo))
+      -- ^ Local SIP state. Includes all revealed SIPs from the
```

Comment:
You are right. But this is at the end of the lifecycle, when the SU is activated. 
Even then, if your delete from the sipdb, you are deleting the update history of the blockchain. Because it is the metadata that provide context to the software update. In the blockchain you only have the transactions that verify how this update got through the lifecycle and managed to get activated. If you loose the metadata, then you have no idea what it was all about. 

Maybe we could issue a final transaction, on activation that will record some minimum metadata for the activated SIP ...